### ISSUE TITLE
Cannot use subapps with overlapped prefixes

### ISSUE THREAD

**Issue by Mirraz** [aio-libs/aiohttp#3701]
## Long story short

If I add to main app one subapp with prefix "/qwe/" and then add another subapp with prefix "/qwerty/" then routing to second subapp does not work. But if I firstly add subapp with prefix "/qwerty/" and then subapp with prefix "/qwe/" then everything works fine. See example.

## Steps to reproduce

```python
from aiohttp.web import Application, json_response, run_app


def create_subapp01():
    app = Application()
    app.router.add_route('GET', r'/aaa', lambda req: json_response({'subapp01': 'aaa'}))
    return app


def create_subapp02():
    app = Application()
    app.router.add_route('GET', r'/aaa', lambda req: json_response({'subapp02': 'aaa'}))
    return app


def create_app():
    app = Application()

    app.add_subapp('/qwe/', create_subapp01())
    app.add_subapp('/qwerty/', create_subapp02())
    # GET /qwerty/aaa -> HTTP code 404

    # app.add_subapp('/qwerty/', create_subapp02())
    # app.add_subapp('/qwe/', create_subapp01())
    # GET /qwerty/aaa -> HTTP code 200

    return app


run_app(create_app(), host='127.0.0.1', port=8080)
```

## Your environment

aiohttp 3.5.4

**Comment by asvetlov** [comment]
This is by design: URL dispatcher handles all routes in the order of adding.
The same behavior is for overlapping routes in the application, not for sub-apps only.

Nothing to do here, the issue is not a bug

**Comment by socketpair** [comment]
@asvetlov what about **slash** ? if We add

```
app.add_subapp('/qwe/', create_subapp01())
app.add_subapp('/qwerty/', create_subapp02())
```

Why queries like `/qwerty/asd` match the first subapp ?

**Comment by asvetlov** [comment]
Ah, I see. That's interesting.

**Comment by nikie** [comment]
Hi, 
Feels like a bug. I have just created PR #3708 which supposedly fixes it.
This is my first contribution, so please be picky and cautious :)

Eugene.

**Comment by asvetlov** [comment]
Fixed by #3708
